File: android/fiscal-cidadao-app/app/src/main/java/br/com/gigascorp/ficalcidadao/modelo/Convenio.java
/*
Copyright (C) 2016  Andrea Mendonça, Emílio Weba, Guilherme Ribeiro, Márcio Oliveira, Thiago Nunes, Wallas Henrique

   This file is part of Fiscal Cidadão.

   Fiscal Cidadão is free software: you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation, either version 3 of the License, or
   (at your option) any later version.

   Fiscal Cidadão is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License
   along with Fiscal Cidadão.  If not, see <http://www.gnu.org/licenses/>.
*/

package br.com.gigascorp.ficalcidadao.modelo;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.Date;

public class Convenio implements Serializable{

    @SerializedName("Id")
    private int id;

    @SerializedName("ObjetoDescricao")
    private String objeto;

    @SerializedName("Latitude")
    private float lat;

    @SerializedName("Longitude")
    private float lng;

    @SerializedName("DataInicio")
    private Date dataInicio;

    @SerializedName("DataFim")
    private Date dataFim;

    @SerializedName("ProponenteNome")
    private String proponente;

    @SerializedName("ProponenteResponsavel")
    private String responsavel;

    @SerializedName("ProponenteTelefone")
    private String telefone;

    @SerializedName("Situacao")
    private String situacao;

    @SerializedName("Valor")
    private float valor;

    public Coordenada getCoordenada(){
        return new Coordenada(this.lat, this.lng);
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getObjeto() {
        return objeto;
    }

    public void setObjeto(String objeto) {
        this.objeto = objeto;
    }

    public float getLat() {
        return lat;
    }

    public void setLat(float lat) {
        this.lat = lat;
    }

    public float getLng() {
        return lng;
    }

    public void setLng(float lng) {
        this.lng = lng;
    }

    public Date getDataInicio() {
        return dataInicio;
    }

    public void setDataInicio(Date dataInicio) {
        this.dataInicio = dataInicio;
    }

    public Date getDataFim() {
        return dataFim;
    }

    public void setDataFim(Date dataFim) {
        this.dataFim = dataFim;
    }

    public String getProponente() {
        return proponente;
    }

    public void setProponente(String proponente) {
        this.proponente = proponente;
    }

    public String getResponsavel() {
        return responsavel;
    }

    public void setResponsavel(String responsavel) {
        this.responsavel = responsavel;
    }

    public String getTelefone() {
        return telefone;
    }

    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }

    public String getSituacao() {
        return situacao;
    }

    public void setSituacao(String situacao) {
        this.situacao = situacao;
    }

    public float getValor() {
        return valor;
    }

    public void setValor(float valor) {
        this.valor = valor;
    }
}

File: ios/FiscalCidadao/Convenio.swift
/*
 Copyright (C) 2016  Andrea Mendonça, Emílio Weba, Guilherme Ribeiro, Márcio Oliveira, Thiago Nunes, Wallas Henrique
 
 This file is part of Fiscal Cidadão.
 
 Fiscal Cidadão is free software: you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation, either version 3 of the License, or
 (at your option) any later version.
 
 Fiscal Cidadão is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.
 
 You should have received a copy of the GNU General Public License
 along with Fiscal Cidadão.  If not, see <http://www.gnu.org/licenses/>.
 */

import UIKit

class Convenio: NSObject
{
    var id : Int
    var startDate : String?
    var endDate : String?
    var location : (Double, Double)
    var desc : String
    var responsible : String
    var proponent : String
    var responsiblePhone : String?
    var status : String?
    var value : Float
    
    init(id : Int, startDate : String?, endDate : String?, location : (Double, Double), desc : String, proponent : String, responsible : String, responsiblePhone : String?, status : String?, value : Float)
    {
        self.id = id
        self.startDate = startDate
        self.endDate = endDate
        self.location = location
        self.desc = desc
        self.proponent = proponent
        self.responsible = responsible
        self.responsiblePhone = responsiblePhone
        self.status = status
        self.value = value
    }
    
    override var description : String
    {
        return "Convenio {id: \(id)" + ", description \(desc), proponente : \(responsible) location(\(location))}"
    }
    
    

}
